src/core/core.hpp

```diff
@@ -22,3 +22,9 @@ struct User {
 std::vector<Product> getProducts(const json& jsonData);
 
 std::vector<User> getUsers(const json& jsonData);
+
+class Authenticator {
```

Comment:
Consider moving `Authenticator` class to authenticator.hpp/cpp
In case you decide to make it under another commit please make an issue out of this